### DISCUSSION TITLE
Is it possible to have hardhat test results output in a parse-able format?

### DISCUSSION THREAD

**Discussion by billtorla-scopely** [NomicFoundation/hardhat#3206] [Troubleshooting] [answered]
I'd like to get hardhat test results as a file I could publish as test results to a continuous integration system. Ideally, a pseudo-standard format like junit xml would be great, but anything parse-able would do (e.g. json, yaml, csv). Is this currently possible? Thank you!

**Comment by fvictorio** [comment, marked as answer]
Yes, add this to your hardhat config:

```js
module.exports = {
  // rest of your config
  mocha: {
    reporter: "json"
  }
}
```

You can use any [mocha reporter](https://mochajs.org/#reporters). I think `xunit` is the one you want.

**Reply by billtorla-scopely** [reply]
Thank you! That works perfectly
